### PR TITLE
Give bazel rules the correct surface

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

**Comment by chrisdunelm** [comment]
Bazel rule surface should now be correct.

Platform and dependency issues (e.g. generator version) will be handled in a future PR.

File: bazel_example/BUILD.bazel [hunk 1]
```diff
@@ -1,4 +1,5 @@
 load("@com_googleapis_gapic_generator_csharp//rules_csharp_gapic:csharp_gapic.bzl", "csharp_proto_library", "csharp_grpc_library", "csharp_gapic_library")
+load("@com_googleapis_gapic_generator_csharp//rules_csharp_gapic:csharp_gapic_pkg.bzl", "csharp_gapic_assembly_pkg")
 
 proto_library(
     name = "example_src",
```
Comment (vam-google) [inline bazel_example/BUILD.bazel:5]: I should have noticed naming like this in the original reveiw (sorry)
According to bazel conventions `proto_library` targets must be called `"<api>_proto"`. Please rename this to `example_proto`

File: bazel_example/BUILD.bazel [hunk 2]
```diff
@@ -9,7 +10,7 @@ proto_library(
 
 csharp_proto_library(
     name = "example_proto",
```
Comment (vam-google) [inline bazel_example/BUILD.bazel:12]: Please rename this to `"example_csharp_proto"` according to bazel naming conventions

File: bazel_example/BUILD.bazel [hunk 3]
```diff
@@ -19,14 +20,27 @@ csharp_grpc_library(
     srcs = [
```
Comment (vam-google) [inline bazel_example/BUILD.bazel:20]: Pleaes rename this target name to `"example_csharp_grpc"` according to bazel conventions

File: bazel_example/BUILD.bazel [hunk 4]
```diff
@@ -19,14 +20,27 @@ csharp_grpc_library(
...
         ":example_src",
     ],
+    deps = [
+        ":example_proto",
+    ]
 )
 
-# This also creates a "example_gapic_full" target, which will contain all required source, not just gapic source.
 csharp_gapic_library(
     name = "example_gapic",
```
Comment (vam-google) [inline bazel_example/BUILD.bazel:29]: Please rename this target to `"example_csharp_gapic"` according to bazel naming conventions.

File: rules_csharp_gapic/csharp_gapic_pkg.bzl [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+_csharp_gapic_assembly_pkg = rule(
+    implementation = _csharp_gapic_assembly_pkg_impl,
+    attrs = {
+        "deps": attr.label_list(mandatory = True, allow_files = True),
+        "package_dir": attr.string(mandatory = True),
+        "_zipper": attr.label(default = Label("@bazel_tools//tools/zip:zipper"), cfg = "host", executable = True),
+    }
+)
+
+def csharp_gapic_assembly_pkg(name, deps, **kwargs):
```
Comment (vam-google) [inline rules_csharp_gapic/csharp_gapic_pkg.bzl:60]: Minor: if a macro does not add anything, it is ok just make the rule itself public (i.e. make `csharp_gapic_assembly_pkg` the name of the rule directly and remove macro)

File: bazel_example/BUILD.bazel (hunk 1 above)
Comment (chrisdunelm) [reply bazel_example/BUILD.bazel]: done

File: bazel_example/BUILD.bazel (hunk 2 above)
Comment (chrisdunelm) [reply bazel_example/BUILD.bazel]: done

File: bazel_example/BUILD.bazel (hunk 3 above)
Comment (chrisdunelm) [reply bazel_example/BUILD.bazel:20]: done

File: bazel_example/BUILD.bazel (hunk 4 above)
Comment (chrisdunelm) [reply bazel_example/BUILD.bazel]: done

File: rules_csharp_gapic/csharp_gapic_pkg.bzl (hunk 5 above)
Comment (chrisdunelm) [reply rules_csharp_gapic/csharp_gapic_pkg.bzl:60]: ack, although I'll leave it as-is for now.

**Pull request merged**